pkg/kubelet/cri/remote/remote_image.go

```diff
@@ -53,35 +58,91 @@ func NewRemoteImageService(endpoint string, connectionTimeout time.Duration) (in
 		return nil, err
 	}
 
-	return &remoteImageService{
-		timeout:     connectionTimeout,
-		imageClient: runtimeapi.NewImageServiceClient(conn),
-	}, nil
+	service := &remoteImageService{
+		timeout:             connectionTimeout,
+		imageClientV1:       runtimeapiV1.NewImageServiceClient(conn),
+		imageClientV1alpha2: runtimeapiV1alpha2.NewImageServiceClient(conn),
+	}
+
+	if err := service.determineAPIVersion(); err != nil {
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func (r *remoteImageService) determineAPIVersion() error {
+	ctx, cancel := getContextWithTimeout(r.timeout)
+	defer cancel()
+
+	if _, err := r.imageClientV1.ListImages(
```

Comment:
> and if have a chosen version.. we can just switch to using the chosen version and let it error out when/if the versioned api is not presen

This works the same way as before, where the runtime complains that an API is unimplemented on the first RPC call. So it would result in a broken node setup before and after this change.